### ISSUE TITLE
Not MinIO compatible?

### ISSUE THREAD

**Issue by LeagueRaINi** [stv0g/gose#21]
So either i'm doing something terrible wrong or gose is not compatible with the latest version of MinIO.

Gose is throwing me this error on startup
```
gose_1   | 2022/06/02 22:31:33 Failed to setup servers: failed to set bucket gose-uploads's CORS rules: MalformedXML: The XML you provided was not well-formed or did not validate against our published schema.
gose_1   |      status code: 400, request id: 16F4EE6811CC0219, host id:
```
and MinIO complains about
```
minio_1  | API: SYSTEM()
minio_1  | Time: 22:31:33 UTC 06/02/2022
minio_1  | DeploymentID: 6fd95624-9ba0-4420-9c8e-e97c46253cf9
minio_1  | Error: expected element type <CreateBucketConfiguration> but have <CORSConfiguration> (xml.UnmarshalError)
minio_1  |        6: internal/logger/logger.go:278:logger.LogIf()
minio_1  |        5: cmd/handler-utils.go:55:cmd.parseLocationConstraint()
minio_1  |        4: cmd/auth-handler.go:344:cmd.checkRequestAuthTypeCredential()
minio_1  |        3: cmd/auth-handler.go:296:cmd.checkRequestAuthType()
minio_1  |        2: cmd/bucket-handlers.go:731:cmd.objectAPIHandlers.PutBucketHandler()
minio_1  |        1: net/http/server.go:2047:http.HandlerFunc.ServeHTTP()
```

**Comment by stv0g** [comment]
Thanks @LeagueRaINi for the Bug report. Gose has indeed only been testen against the Ceph RadosGW S3 Implementation (see also #19).

I will have a look soon to fix it also for Minio. Contributions are also highly welcome.

**Comment by stv0g** [comment]
Thanks @LeagueRaINi for the Bug report. Gose has indeed only been testen against the Ceph RadosGW S3 Implementation (see also #19).

I will have a look soon to fix it also for Minio. Contributions are also highly welcome.

**Comment by stv0g** [comment]
I started debugging the issue and found the root cause:

Gose is issuing a `PutBucketCors` request which is not supported by Minio:

> List of Amazon S3 Bucket API's not supported on MinIO
> - BucketCORS (CORS enabled by default on all buckets for all HTTP verbs)

(https://docs.min.io/docs/minio-server-limits-per-tenant.html)

As a fix, I will add some detection of the underlying S3 implementation and disable the unsupported features.

**Comment by stv0g** [comment]
Should now be fixed in v0.4.0
